### PR TITLE
nuc-165 move lock implementation from gopher into rdbi

### PR DESCRIPTION
**Pull request by KhaiNguyen**

### CODE REVIEW

File: src/main/java/com/lithium/dbi/rdbi/recipes/locking/MultiReadWriteLock.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
+package com.lithium.dbi.rdbi.recipes.locking;
+
+import com.google.common.base.Throwables;
+import com.lithium.dbi.rdbi.Handle;
+import com.lithium.dbi.rdbi.RDBI;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+
+public class MultiReadWriteLock {
```
Comment (kumarad) [inline src/main/java/com/lithium/dbi/rdbi/recipes/locking/MultiReadWriteLock.java]: do we need to be more explicit that this is a multi read single write lock ?

File: src/main/java/com/lithium/dbi/rdbi/recipes/locking/MultiReadWriteLock.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            while (true) {
+                final int acquiredLock = dao.acquireWriteLock(writeLockKey, ownerId, writeTimeoutMillis);
+                if (acquiredLock > 0) {
+                    break;
+                }
+
+                Thread.sleep(250);
+            }
+
+            while (true) {
```
Comment (kumarad) [inline src/main/java/com/lithium/dbi/rdbi/recipes/locking/MultiReadWriteLock.java]: might be worth adding a comment about what case you are addressing here. id imagine others will wonder why you acquire the write lock and then wait for the read lock to get acquired like i did

**Comment by kumarad** [comment]
lgtm

File: src/main/java/com/lithium/dbi/rdbi/recipes/locking/MultiReadWriteLock.java (hunk 1 above)
Comment (KhaiNguyen) [inline src/main/java/com/lithium/dbi/rdbi/recipes/locking/MultiReadWriteLock.java]: I was thinking that earlier... I can rename

File: src/main/java/com/lithium/dbi/rdbi/recipes/locking/MultiReadWriteLock.java (hunk 2 above)
Comment (KhaiNguyen) [inline src/main/java/com/lithium/dbi/rdbi/recipes/locking/MultiReadWriteLock.java]: yeah, can do

**Pull request merged**